src/js/jsx/shared/Datalist.jsx

```diff
@@ -383,12 +391,14 @@ define(function (require, exports, module) {
                 
                 width += 20 * iconCount; // 20 pixels is the computed width + padding of an svg icon
 
+                React.findDOMNode(this.refs.autocomplete).style.left = width + "px";
```

Comment:
Yikes, this is pretty scary. It doesn't seem like this is something that should exist inside the Datalist component. Ideally the width of the menu would expand to accommodate the widths of all the entries. I'm sure there are lot of challenges with this in CSS though. Can you explain what's going on here?